File: code_samples/elixir/performance_lazy_loading.ex
# Performance: Lazy Loading
# AI/ML Training Sample

defmodule LazyLoading do
  defstruct data: ""
  
  def new(), do: %LazyLoading{}
  
  def process(%LazyLoading{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%LazyLoading{data: data}), do: data
  
  def validate(%LazyLoading{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = LazyLoading.new()
updated = LazyLoading.process(instance, "example")
IO.puts("Data: " <> LazyLoading.get_data(updated))
IO.puts("Valid: " <> to_string(LazyLoading.validate(updated)))

File: code_samples/go/performance_lazy_loading.go
package main

import (
    "fmt"
)

// Performance: Lazy Loading
// AI/ML Training Sample

type LazyLoading struct {
    Data string
}

func NewLazyLoading() *LazyLoading {
    return &LazyLoading{
        Data: "",
    }
}

func (s *LazyLoading) Process(input string) {
    s.Data = input
}

func (s *LazyLoading) Validate() bool {
    return len(s.Data) > 0
}

func (s *LazyLoading) GetData() string {
    return s.Data
}

func main() {
    instance := NewLazyLoading()
    instance.Process("example")
    fmt.Printf("Data: %s\n", instance.GetData())
    fmt.Printf("Valid: %v\n", instance.Validate())
}
